staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -94,27 +97,26 @@ func getResourceHandler(scope RequestScope, getter getterFunc) http.HandlerFunc
 		ctx := scope.ContextFunc(req)
 		ctx = request.WithNamespace(ctx, namespace)
 
-		result, err := getter(ctx, name, req)
+		result, err := getter(ctx, name, req, trace)
 		if err != nil {
 			scope.err(err, w, req)
 			return
 		}
+		trace.Step("About to create self-link")
```

Comment:
Setting selflink for just single object should be super fast. In case of lists we have it just because it may take time to self link e.g. 10.000 objects.

So I think that creating this string (and GC-ing it) will cost more than it's worth it.